### ISSUE TITLE
Training SSD on KITTI dataset - loss diverges (loss = nan)

### ISSUE THREAD

**Issue by neermat** [weiliu89/caffe#128]
Hi,

I am trying to train the network on KITTI dataset for pedestrian detection but the network loss quickly diverges to 'nan':

> I0821 13:09:56.750145 19638 solver.cpp:231] Iteration 0, loss = 21.6547
> I0821 13:09:56.750180 19638 solver.cpp:247]     Train net output #0: mbox_loss = 21.6547 (* 1 = 21.6547 loss)
> I0821 13:09:57.298372 19638 sgd_solver.cpp:106] Iteration 0, lr = 0.001
> I0821 13:10:09.782060 19638 solver.cpp:231] Iteration 10, loss = 92.9826
> I0821 13:10:09.782094 19638 solver.cpp:247]     Train net output #0: mbox_loss = 663.16 (* 1 = 663.16 loss)
> I0821 13:10:10.232007 19638 sgd_solver.cpp:106] Iteration 10, lr = 0.001
> I0821 13:10:22.536900 19638 solver.cpp:231] Iteration 20, loss = nan
> I0821 13:10:22.536931 19638 solver.cpp:247]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
> I0821 13:10:22.994793 19638 sgd_solver.cpp:106] Iteration 20, lr = 0.001
> 

The following is a list of all the changes I made prior to running `ssd_pascal.py`:

- Created `labelmap_kitti.prototxt`:
```
item {
name: "none_of_the_above"
label: 0
display_name: "background"
}
item {
name: "person"
label: 1
display_name: "person"
}
```
- Converted KITTI annotations to format: `1 712.40 143.00 810.73 307.92`. Since I'm only interested in pedestrian detection, some of the `.txt` files are empty (Is this a problem?).
- Executed create_list.sh and create_data.sh with changes: `extra_cmd="--encode-type=png --encoded"` and `--label-type="txt"`.
- Changed `num_classes` from 21 to 2 in ssd_pascal.py and gave new paths to train and test data.

 Am I missing something? Any help is appreciated.

**Comment by weiliu89** [comment]
You could try lowering your initial learning rate

**Comment by neermat** [comment]
Thanks. I should have tried that earlier.

**Comment by catsdogone** [comment]
@neermat How do you set the image height and with when training and testing? Just keep the original image size or set it to 300 or 500?

**Comment by neermat** [comment]
@catsdogone For both training and testing, I kept original aspect ratio of KITTI images. 546x165 (SSD300) and 910x275 (SSD500)

**Comment by catsdogone** [comment]
@neermat Thank you very much for your information.

**Comment by XhqGlorry11** [comment]
@neermat Did you solve the problem of nan loss after lowering initial learning rate?

**Comment by XhqGlorry11** [comment]
@neermat hello, did you solve the problem of loss nan by lowering the initial learning rate? And did you succeeded in training ssd model with KITTI for pedestrian detection?

**Comment by neermat** [comment]
@XhqGlorry11 yes, lowering the learning rate by a factor of two for first few thousand iterations helped. And yes, I could finally train an SSD model for car and pedestrian detection on KITTI images.

**Comment by XhqGlorry11** [comment]
@Got it! Thank you so much for your quick reply.

**Comment by XhqGlorry11** [comment]
@neermat Got it! Thank you so much for your quick reply.

**Comment by XhqGlorry11** [comment]
@neermat Hi~Do you keep your log file of training ssd model with KITTI? I am training ssd model with voc data, but I find the loss keeps staying around 5~6 even after approximately 10000 iteration. I want to find out the changes of loss by log file.

**Comment by XhqGlorry11** [comment]
@neermat Hi! Do you keep your log file of training ssd model with KITTI? I am training ssd model with voc data, but I find the loss keeps staying around 56 even after approximately 10000 iteration. I want to find out the changes of loss by log file.

**Comment by XhqGlorry11** [comment]
@neermat Hi! Do you keep your log file of training ssd model with KITTI? I am training ssd model with voc data, but I find the loss keeps staying around 5~6 even after approximately 10000 iteration. I want to find out the changes of loss by log file.

**Comment by MarcosPieras** [comment]
Hi @neermat @XhqGlorry11 , I am working on people detection too. Did you notice any improvement in perfomance with respect to models trained with PASCAL or COCO ?

**Comment by IEEE-FELLOW** [comment]
hi @XhqGlorry11 @MarcosPieras  Have u trained ur model?I train ssd on my dataset and got loss around 3 after 2400 iterations. I wonder what's the final loss should be?
